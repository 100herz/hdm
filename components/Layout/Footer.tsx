import Link from "next/link"
import styled from "styled-components"

const Address = styled.address`
  font-size: 1.125rem;
  font-style: normal;
  line-height: 2.5rem;
`

type Props = {
  className?: string
}

const Footer: React.FC<Props> = ({ className = "" }: Props) => {
  return (
    <footer className={`container px-6  ${className}`}>
      <section>
        <h2 className="py-16 text-4xl font-bold leading-10">
          Nehmen Sie <br />
          Kontakt mit <br />
          uns auf
        </h2>

        <Address>
          Holz Design Moers GmbH <br />
          Am Schürmannshütt 23 <br />
          47441 Moers
        </Address>

        <Address className="mt-6">
          Telefon:{" "}
          <a className="underline" href="tel:+49284114010">
            +49 2841 1401 0
          </a>{" "}
          <br />
          Telefax: +49 2841 1401 180 <br />
          E-Mail:{" "}
          <a className="underline" href="mailto:info@hdm.gmbh">
            info@hdm.gmbh
          </a>
        </Address>

        <div className="mt-24">
          <img className="w-48" src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" />
        </div>
      </section>

      <section className="mt-6 text-lg leading-10">
        © 2020 Holz Design Moers GmbH
        <nav className="mt-2">
          <Link href="/imprint" as="impressum">
            <a>Impressum</a>
          </Link>
          <Link href="/privacy" as="datenschutz">
            <a className="ml-8">Datenschutz</a>
          </Link>
        </nav>
      </section>

      <div className="mt-10 text-right">
        <button>
          <svg
            className="inline-block w-6 transform -rotate-90"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24.242 45"
          >
            <path
              d="M126.031,21.266,105.252.486A1.732,1.732,0,0,0,102.8,2.935L122.359,22.49,102.8,42.045a1.731,1.731,0,0,0,2.448,2.448l20.779-20.779A1.732,1.732,0,0,0,126.031,21.266Z"
              transform="translate(-102.297 0)"
              fill="#0084b0"
            />
          </svg>
        </button>
      </div>
    </footer>
  )
}

export default Footer
