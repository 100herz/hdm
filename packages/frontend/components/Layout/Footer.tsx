import Link from 'next/link'
import Image from 'next/image'
import { useRouter } from 'next/router'
import { animateScroll } from 'react-scroll'
import styled from '@emotion/styled'

export default function Footer({ className = '' }: React.HTMLAttributes<HTMLElement>) {
  const router = useRouter()

  return (
    <FooterStyled id="kontakt" className={`lg:mt-40 pb-10 ${className}`}>
      <section className="container lg:flex lg:justify-between lg:space-x-6">
        <h2 className="py-16 text-4xl font-bold leading-10 lg:pt-0">
          Nehmen Sie <br />
          Kontakt mit <br />
          uns auf
        </h2>

        <Address>
          Holz Design Moers GmbH <br />
          Am Schürmannshütt 23 <br />
          47441 Moers
        </Address>

        <Address className="mt-6 lg:mt-0">
          Telefon:{' '}
          <a className="underline hover:text-hdm-dark" href="tel:+4928411401101">
            +49 2841 1401 101
          </a>
          <br />
          Telefax: +49 2841 1401 180 <br />
          E-Mail:{' '}
          <a className="underline hover:text-hdm-dark" href="mailto:info@hdm.gmbh">
            info@hdm.gmbh
          </a>
        </Address>

        <div className="mt-24 lg:mt-0">
          {router && router.pathname === '/' ? (
            <button
              className="w-48"
              id="logo"
              onClick={() => animateScroll.scrollToTop({ duration: 800 })}
              type="button"
              aria-label="HDM Logo"
            >
              <Image src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" width={192} height={49} />
            </button>
          ) : (
            <Link href="/">
              <a className="w-48">
                <Image src="/ci/hdm-logo.svg" alt="Logo der HDM GmbH" width={192} height={49} />
              </a>
            </Link>
          )}
        </div>
      </section>

      <section className="container mt-6 text-lg leading-10 lg:flex lg:justify-center">
        <p>&copy; 2020 Holz Design Moers GmbH</p>
        <nav className="flex flex-col mt-2 leading-8 lg:mt-0 lg:flex-row">
          <Link href="/downloads">
            <a className="underline lg:ml-8 hover:text-hdm-dark">Downloads</a>
          </Link>
          <Link href="/impressum">
            <a className="underline lg:ml-8 hover:text-hdm-dark">Impressum</a>
          </Link>
          <Link href="/datenschutz">
            <a className="underline lg:ml-8 hover:text-hdm-dark">Datenschutz</a>
          </Link>
        </nav>
      </section>

      <div className="container my-10 text-right lg:text-center">
        <button
          type="button"
          id="back-to-top-button"
          onClick={() => animateScroll.scrollToTop({ duration: 800 })}
          aria-label="Go to top"
        >
          <svg
            className="inline-block w-6 transform -rotate-90"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24.242 45"
          >
            <path
              d="M126.031,21.266,105.252.486A1.732,1.732,0,0,0,102.8,2.935L122.359,22.49,102.8,42.045a1.731,1.731,0,0,0,2.448,2.448l20.779-20.779A1.732,1.732,0,0,0,126.031,21.266Z"
              transform="translate(-102.297 0)"
              fill="#0084b0"
            />
          </svg>
        </button>
      </div>
    </FooterStyled>
  )
}

const FooterStyled = styled.footer`
  position: relative;

  @media (min-width: 1024px) {
    background-image: linear-gradient(rgba(255, 255, 255, 0.95), rgba(255, 255, 255, 0.95)), url(/ci/hdm-bg.svg);
    background-repeat: no-repeat;
    background-size: 100% auto;
  }
`

const Address = styled.address`
  font-size: 1.125rem;
  font-style: normal;
  line-height: 2.5rem;
`
