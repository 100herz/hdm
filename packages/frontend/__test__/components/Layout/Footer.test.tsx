import Footer from '@/components/Layout/Footer'
import { withTestRouter } from '@/__test__/__hoc__/withTestRouter'
import { mount, shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

describe('Footer Testings', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<Footer />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  test('render one button component with the id #logo', () => {
    const wrapper = mount(withTestRouter(<Footer />, { pathname: '/' }))
    const button = wrapper.find('#logo')
    button.simulate('click')
    expect(button).toHaveLength(1)
  })

  test("don't render an button component with the id #logo", () => {
    const wrapper = mount(withTestRouter(<Footer />, { pathname: '/abc' }))
    expect(wrapper.find('#logo')).toHaveLength(0)
  })

  test('simulate a click on the button with he id #back-to-top-button', () => {
    const wrapper = mount(withTestRouter(<Footer />, { pathname: '/' }))
    const button = wrapper.find('#back-to-top-button')
    button.simulate('click')
    expect(button).toHaveLength(1)
  })
})
