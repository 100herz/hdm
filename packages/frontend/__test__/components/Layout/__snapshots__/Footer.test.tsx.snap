// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Testings matches snapshot 1`] = `
<FooterStyled
  className="lg:mt-40 pb-10 "
  id="kontakt"
>
  <section
    className="container lg:flex lg:justify-between lg:space-x-6"
  >
    <h2
      className="py-16 text-4xl font-bold leading-10 lg:pt-0"
    >
      Nehmen Sie 
      <br />
      Kontakt mit 
      <br />
      uns auf
    </h2>
    <Address>
      Holz Design Moers GmbH 
      <br />
      Am Schürmannshütt 23 
      <br />
      47441 Moers
    </Address>
    <Address
      className="mt-6 lg:mt-0"
    >
      Telefon:
       
      <a
        className="underline hover:text-hdm-dark"
        href="tel:+4928411401101"
      >
        +49 2841 1401 101
      </a>
      <br />
      Telefax: +49 2841 1401 180 
      <br />
      E-Mail:
       
      <a
        className="underline hover:text-hdm-dark"
        href="mailto:info@hdm.gmbh"
      >
        info@hdm.gmbh
      </a>
    </Address>
    <div
      className="mt-24 lg:mt-0"
    >
      <Link
        href="/"
      >
        <a>
          <img
            alt="Logo der HDM GmbH"
            className="w-48"
            src="/ci/hdm-logo.svg"
          />
        </a>
      </Link>
    </div>
  </section>
  <section
    className="container mt-6 text-lg leading-10 lg:flex lg:justify-center"
  >
    <p>
      © 2020 Holz Design Moers GmbH
    </p>
    <nav
      className="mt-2 leading-8 lg:mt-0"
    >
      <Link
        href="/impressum"
      >
        <a
          className="underline hover:text-hdm-dark lg:ml-8"
        >
          Impressum
        </a>
      </Link>
      <Link
        href="/datenschutz"
      >
        <a
          className="ml-8 underline hover:text-hdm-dark"
        >
          Datenschutz
        </a>
      </Link>
    </nav>
  </section>
  <div
    className="container my-10 text-right lg:text-center"
  >
    <button
      id="back-to-top-button"
      onClick={[Function]}
    >
      <svg
        className="inline-block w-6 transform -rotate-90"
        viewBox="0 0 24.242 45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M126.031,21.266,105.252.486A1.732,1.732,0,0,0,102.8,2.935L122.359,22.49,102.8,42.045a1.731,1.731,0,0,0,2.448,2.448l20.779-20.779A1.732,1.732,0,0,0,126.031,21.266Z"
          fill="#0084b0"
          transform="translate(-102.297 0)"
        />
      </svg>
    </button>
  </div>
</FooterStyled>
`;
